pkg/gui/controllers/branches_controller.go

```diff
@@ -571,43 +584,38 @@ func (self *BranchesController) fastForward(branch *models.Branch) error {
 	)
 
 	return self.c.WithWaitingStatus(message, func(task gocui.Task) error {
-		worktree, ok := self.worktreeForBranch(branch)
-		if ok {
-			self.c.LogAction(action)
-
-			worktreeGitDir := ""
-			// if it is the current worktree path, no need to specify the path
-			if !worktree.IsCurrent {
-				worktreeGitDir = worktree.GitDir
-			}
-
-			err := self.c.Git().Sync.Pull(
-				task,
-				git_commands.PullOptions{
-					RemoteName:      branch.UpstreamRemote,
-					BranchName:      branch.UpstreamBranch,
-					FastForwardOnly: true,
-					WorktreeGitDir:  worktreeGitDir,
-				},
-			)
-			if err != nil {
-				_ = self.c.Error(err)
-			}
+		return self.c.State().WithItemOperation(branch, types.ItemOperationFastForwarding, context.LOCAL_BRANCHES_CONTEXT_KEY, func() error {
```

Comment:
Can we add WithItemOperation to pkg/gui/gui_common.go? Feels like it should be just as accessible as WithWaitingStatus

Reply:
You mean like this? ca6fefd0ccc0

I'm still a bit unsure sometimes with all the interfaces that we have. WithWaitingStatus is part of IPopupHandler (not really sure why to be honest), for WithItemOperation I couldn't think of anything better than IGuiCommon.

Reply:
yep!